src/generic-container/generic-container.ts

```diff
@@ -203,10 +206,18 @@ export class GenericContainer implements TestContainer {
       this.waitStrategy.withStartupTimeout(this.startupTimeout);
     }
 
-    if (containerLog.enabled()) {
-      (await containerLogs(container))
-        .on("data", (data) => containerLog.trace(`${container.id}: ${data.trim()}`))
-        .on("err", (data) => containerLog.error(`${container.id}: ${data.trim()}`));
+    if (containerLog.enabled() || this.logConsumer !== undefined) {
+      const logStream = await containerLogs(container);
+
+      if (this.logConsumer !== undefined) {
+        this.logConsumer(logStream);
+      }
+
+      if (containerLog.enabled()) {
+        logStream
+          .on("data", (data) => containerLog.trace(data.trim(), { containerId: container.id }))
```

Comment:
Will be nice to have containerName/imageName here (one of, name > image), as we discussed in my PR

Reply:
Actually see this comment on your PR where I explained why not to use the container name: https://github.com/testcontainers/testcontainers-node/pull/558#issuecomment-1536068467